projects/plugins/beta/jetpack-beta.php

```diff
@@ -57,1183 +59,14 @@
-			if ( ! file_exists( $source . 'jetpack.php' ) ) {
-				return new WP_Error( 'plugin_file_does_not_exist', __( 'Main Plugin File does not exist', 'jetpack-beta' ) );
-			}
-			if ( ! file_exists( $source . '_inc/build/static.html' ) ) {
-				return new WP_Error( 'static_admin_page_does_not_exist', __( 'Static Admin Page File does not exist', 'jetpack-beta' ) );
-			}
-			if ( ! file_exists( $source . '_inc/build/admin.js' ) ) {
-				return new WP_Error( 'admin_page_does_not_exist', __( 'Admin Page File does not exist', 'jetpack-beta' ) );
-			}
-			// It has happened that sometimes a generated bundle from the master branch ends up with an empty
-			// vendor directory. Used to be a problem in the beta building process.
-			if ( self::is_dir_empty( $source . 'vendor' ) ) {
-				return new WP_Error( 'vendor_dir_is_empty', __( 'The dependencies dir (vendor) is empty', 'jetpack-beta' ) );
-			}
-		}
-
-		return $source;
-	}
-
-	/**
-	 * Checks if a dir is empty
-	 *
-	 * @param [type] $dir The absolute directory path to check
-	 * @return boolean
-	 */
-	static function is_dir_empty( $dir ) {
-		return ( count( scandir( $dir ) ) == 2 );
-	}
-
-	/**
-	 * Callback function to include Jetpack beta options into Jetpack sync whitelist
-	 *
-	 * @param Array $whitelist List of whitelisted options to sync
-	 */
-	public function add_to_options_whitelist( $whitelist ) {
-		$whitelist[] = self::$option;
-		$whitelist[] = self::$option_dev_installed;
-		$whitelist[] = self::$option_autoupdate;
-		$whitelist[] = self::$option_email_notif;
-		return $whitelist;
-	}
-
-	/**
-	 * Custom error handler to intercept errors and log them using Jetpack's own logger.
-	 *
-	 * @param int $errno error code.
-	 * @param string $errstr error message.
-	 * @param string $errfile file name where the error happened.
-	 * @param int $errline line in the code.
-	 *
-	 * @return bool whether to make the default handler handle the error as well.
-	 */
-	public static function custom_error_handler( $errno, $errstr, $errfile, $errline ) {
-
-		if ( class_exists( 'Jetpack' ) && method_exists( 'Jetpack', 'log' ) ) {
-			$error_string = sprintf( "%s, %s:%d", $errstr, $errfile, $errline );
-
-			// Only adding to log if the message is related to Jetpack.
-			if ( false !== stripos( $error_string, 'jetpack' ) ) {
-				Jetpack::log( $errno, $error_string );
-			}
-		}
-
-		/**
-		 * The error_reporting call returns current error reporting level as an integer. Bitwise
-		 * AND lets us determine whether the current error is included in the current error
-		 * reporting level
-		 */
-		if ( ! ( error_reporting() & $errno ) ) {
-
-			// If this error is not being reported in the current settings, stop reporting here by returning true.
-			return true;
-		}
-
-		// Returning false makes the error go through the standard error handler as well.
-		return false;
-	}
-}
+set_error_handler( array( 'Jetpack_Beta', 'custom_error_handler' ) );
```

Comment:
I think this is fine. We can add the `// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_set_error_handler` to the end of this line.

Bigger picture, I'm not sure we need to have this logging at all, BUT, that's outside the scope of getting this into the repo.

Reply:
Squelched in [507b62d](https://github.com/Automattic/jetpack/pull/18405/commits/507b62dbc8eb249b9390991c0b80f0d82b6a80d7)